SYCL/ESIMD/api/functional/functions/functions_1d_select.hpp

```diff
@@ -0,0 +1,281 @@
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> result(NumElems, allocator);
+    shared_vector<DataT> initial_ref_data(NumElems, allocator);
+    shared_vector<DataT> ref_data_for_fill(NumElems, allocator);
+
+    shared_element<bool> is_correct_type(queue, true);
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      initial_ref_data[i] = i + 1;
+    }
+    // We should have different values in the first reference data and in the
+    // second reference data.
+    for (size_t i = 0; i < NumElems; ++i) {
+      ref_data_for_fill[i] =
+          initial_ref_data[i] + value_for_increase_ref_data_for_fill;
+    }
+
+    queue.submit([&](sycl::handler &cgh) {
+      DataT *init_ref_ptr = initial_ref_data.data();
+      DataT *ref_data_for_fill_ptr = ref_data_for_fill.data();
+      DataT *const out_ptr = result.data();
+      auto is_correct_type_ptr = is_correct_type.data();
+
+      cgh.single_task<
+          Kernel<DataT, NumElems, TestCaseT, NumSelectedElemsT, StrideT>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            *is_correct_type_ptr =
+                TestCaseT::template call_operator<DataT, NumElems,
+                                                  NumSelectedElems, Stride>(
+                    init_ref_ptr, ref_data_for_fill_ptr, out_ptr, Offset);
+          });
+    });
+    queue.wait_and_throw();
+
+    std::vector<size_t> selected_indexsess;
+    // Collect the indexess that has been selected.
+    for (size_t i = Offset; i < Stride * NumSelectedElems + Offset;
+         i += Stride) {
+      selected_indexsess.push_back(i);
+    }
+
+    // Push the largest value to avoid the following error: can't dereference
+    // out of range vector iterator.
+    selected_indexsess.push_back(std::numeric_limits<size_t>::max());
+    auto selected_indexsess_ptr = selected_indexsess.begin();
+
+    // Verify that values, that do not was selected has initial values.
+    for (size_t i = 0; i < NumElems; ++i) {
+      // If current index is less than selected index verify that this element
+      // hasn't been selected and changed.
+      if (i < *selected_indexsess_ptr) {
+        DataT expected = initial_ref_data[i];
+        DataT retrieved = result[i];
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type);
+        }
+      } else {
+        DataT expected = ref_data_for_fill[(i - Offset) / Stride];
+        DataT retrieved = result[i];
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type);
+        }
+        selected_indexsess_ptr++;
+      }
+    }
+
+    if (!is_correct_type.value()) {
+      passed = false;
+      log::print_line("Test failed due to type of the object that returns " +
+                      TestCaseT::get_description() +
+                      " operator is not equal to the expected one for simd<" +
+                      data_type + ", " + std::to_string(NumElems) + ">.");
```

Comment:
```suggestion
      log::fail(TestDescriptionT(data_type), "Unexpected return type");
```
Please note that `TestDescriptionT` should have all test case information logged inside.
So, for example,

- if you have `log::trace<TestDescriptionT>(data_type, other arguments);`
- in `run_test::operator()`
- and define `ESIMD_TESTS_VERBOSE_LOG`
- there would be no equal lines

Reply:
The `TestDescription` has been updated in this commit: https://github.com/intel/llvm-test-suite/pull/875/commits/62ef8cdc98df2c8ba5b02911491adc9a7f432d42